### PR TITLE
Add zero conf support

### PR DESCRIPTION
**Pull request by niteshbalusu11**
Signed-off-by: Nitesh Balusu <84944042+niteshbalusu11@users.noreply.github.com>

Added dev dependencies of typescript, concurrently and nodemon

Updated protos to lnd v0.16.4

Added zero conf support.

Channel open flow first attempts zero conf two times with a timeout of 4s.

If zero conf fails, then a regular open is attempted twice with a 4s timeout.

### CODE REVIEW

File: src/utils/lnd-api.ts [hunk 1]
```diff
@@ -75,6 +76,8 @@ export async function openChannelSync(
     minConfs: 0,
     private: privateChannel,
     spendUnconfirmed,
+    zeroConf,
+    commitmentType: 3,
```
Comment (hsjoberg) [inline src/utils/lnd-api.ts:80]:
```suggestion
    commitmentType: lnrpc.CommitmentType.ANCHORS,
```

**Review by hsjoberg — CHANGES_REQUESTED** [review]
Code looks good!

I would to have the ability to disable zero conf openings. Please add a config for it.

See:
`config/default.config_TEMPLATE`
`src/utils/common.ts` (bottom)

**Comment by niteshbalusu11** [comment]
Ok sure.

**Review by hsjoberg — APPROVED** [review]
Looks good to me. ✅ 

Add the config to `config/test.json` too. Also needs rebase.

**Comment by niteshbalusu11** [comment]
Fixed

**Pull request merged**